src/Controller/Controller.php

```diff
@@ -54,8 +54,16 @@ public function get($serviceName)
 	public function addFlash($type, $message, array $params = [])
 	{
 		$message = $this->trans($message, $params);
+		$this->get('http.session')->getFlashBag()->add($type, $message);
 
-		return $this->get('http.session')->getFlashBag()->add($type, $message);
+		// Filter out duplicate messages
+		$flashes = $this->get('http.session')->getFlashBag()->peekAll();
+		foreach ($flashes as $type => $messages) {
```

Comment:
Do we need this foreach? We are adding to `$type` so the only one that will not be unique will be `$flashes[$type]`.

Would it not be better to foreach over `$flashes[$type]` and only add the message if it's not found?